prymer/primer3/primer3.py

```diff
@@ -530,13 +530,27 @@ def _build_oligos(
             bases = unmasked_design_seq[slice_offset:slice_end]
             if span.strand == Strand.NEGATIVE:
                 bases = reverse_complement(bases)
-
+            # assemble Primer3-emitted results into `Oligo` objects
+            # if thermodynamic melting temperatures are missing, default to None
             primers.append(
                 Oligo(
                     bases=bases,
                     tm=float(design_results[f"PRIMER_{design_task.task_type}_{idx}_TM"]),
                     penalty=float(design_results[f"PRIMER_{design_task.task_type}_{idx}_PENALTY"]),
                     span=span,
+                    self_any_th=float(
+                        design_results.get(
+                            f"PRIMER_{design_task.task_type}_{idx}_SELF_ANY_TH", None
+                        )
+                    ),
```

Comment:
In my hands `float(None)` raises an exception.  I think this needs to account for that possibility.

Reply:
Relatedly, should we enforce `PRIMER_THERMODYNAMIC_OLIGO_ALIGNMENT=1` and any other necessary parameters when submitting in `primer3.py::design()` to ensure thermodynamic values are always produced?  It's the default for the latest version, but not all verisons.

Reply:
@msto My suggestion above was to make sure these values are never `None`.  There are parameters that one supplies to primer3 to turn on thermodynamic calculations, and I'm suggesting we should ensure those are always on, in which case these should never be None.

Reply:
Then we should also update the corresponding attributes to be non-`Optional`, correct?